tests/test_rds.py

```diff
@@ -1545,6 +1545,37 @@ def test_rds_consecutive_aws_backups_count_filter(self):
             resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_rds_snapshot_copy_related_tags(self):
+        factory = self.replay_flight_data("test_rds_snapshot_copy_related_tags")
+        client = factory().client("rds")
+        p = self.load_policy(
+            {
+                "name": "rds-snapshot-copy-related-tags",
+                "resource": "rds-snapshot",
+                "filters": [
+                    {"tag:Owner": "absent"},
+                    {"DBSnapshotIdentifier": "rds:database-1-2023-05-31-15-53"}
+                    ],
```

Comment:
I made two changes to this test locally so it would fail without your code changes and pass with them...

```suggestion
```

Reply:
Ah, I see. Yes, the test would be causing only one snapshot to be filtered and we wouldn't hit the issue. I'll go ahead and change the tests so that we able to test the real issue here